src/test/java/spoon/test/main/MainTest.java

```diff
@@ -66,7 +64,21 @@ public void checkGenericContracts(CtPackage pack) {
 
 		// scanners
 		checkContractCtScanner(pack);
-		
+
+		// shadow
+		checkShadow(pack);
+	}
+
+	private void checkShadow(CtPackage pack) {
```

Comment:
I like this, since MainTest is full classpath, we can also assess that getType/Field/ExecutableDeclaration never returns null.